app/models/miq_cockpit_ws_worker.rb

```diff
@@ -0,0 +1,64 @@
+require 'miq_apache'
+
+class MiqCockpitWsWorker < MiqWorker
+  require_nested :Runner
+
+  APACHE_CONF_FILE = '/etc/httpd/conf.d/manageiq-cockpit.conf'.freeze
+  self.required_roles = ['cockpit_ws']
+  self.maximum_workers_count = 1
+
+  def friendly_name
+    @friendly_name ||= "Cockpit Worker"
+  end
+
+  def self.can_start_cockpit_ws?
+    @supports_cockpit_ws ||= MiqEnvironment::Command.is_linux? &&
+                             File.exist?('/usr/libexec/cockpit-ws') &&
+                             File.exist?('/usr/libexec/cockpit-ssh')
+  end
+
+  def self.should_start_worker?
+    return false unless has_required_role?
+    can_start_cockpit_ws?
+  end
+
+  def self.sync_workers
+    opts = MiqCockpitWsWorker.worker_settings
+    install_apache_proxy_config(opts) if MiqEnvironment::Command.supports_apache?
+    @workers = should_start_worker? ? 1 : 0
+    super
+  end
+
+  def self.install_apache_proxy_config(opts = nil)
+    opts ||= MiqCockpitWsWorker.worker_settings
+    if has_required_role? || opts[:external_url]
+      url = MiqCockpit::WS.url(APACHE_CONF_FILE)
+    elsif MiqServer.my_server.has_active_role?(MiqUiWorker.required_roles)
+      url = MiqRegion.my_region.remote_cockpit_ws_url
+    end
+
+    if url
+      MiqCockpit::ApacheConfig.new(url).save(APACHE_CONF_FILE)
```

Comment:
I believe this is called from `sync_workers` so it should be ok, but if the `cockpit_ws` role fails over to another appliance, I want to make sure that any configuration changes will also happen on the new `cockpit_ws` appliance.